File: laundry/wash_wallet_be/resources/js/types/category.ts
import { BaseFilters, BaseSortOptions, LaundryService, Outlet } from ".";

export interface Category {
    id: number;
    outletId: number;
    name: string;
    description?: string | null;
    slug: string;
    isActive: boolean;
    createdAt: string;
    updatedAt: string;
    deletedAt?: string | null;
    outlet: Outlet;
    laundryServices?: LaundryService[];
    laundryServicesCount?: number;
    statusLabel?: string;
}

/**
 * Category filter interface
 */
export interface CategoryFilters extends BaseFilters {
    isActive?: boolean;
    outletId?: number;
}

/**
 * Category sort options
 */
export interface CategorySortOptions extends BaseSortOptions {
    column:
        | "name"
        | "slug"
        | "description"
        | "isActive"
        | "createdAt"
        | "updatedAt";
}

/**
 *  Category form data
 */
export interface CategoryFormData {
    outletId: number;
    name: string;
    description?: string;
    isActive?: boolean | null;
}

export interface CategoryLaundryServiceFormData {
    unitId: number;
    name: string;
    description?: string | null;
    price: number;
    durationHours: number;
    minQuantity: number;
    laundryServiceProcesses?: Array<{
        processId: number;
    }>;
}

File: laundry/wash_wallet/packages/wash_wallet_domain/lib/src/entities/category.dart
import 'package:equatable/equatable.dart';
import 'laundry_service.dart';

class Category extends Equatable {
  final int id;
  final int? outletId;
  final String name;
  final String slug;
  final String? description;
  final bool isActive;
  final int laundryServicesCount;
  final List<LaundryService>? laundryServices;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final DateTime? deletedAt;

  const Category({
    required this.id,
    this.outletId,
    required this.name,
    required this.slug,
    this.description,
    required this.isActive,
    this.laundryServicesCount = 0,
    this.laundryServices,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
  });

  factory Category.fromModel(dynamic model) {
    return Category(
      id: model.id,
      outletId: model.outletId,
      name: model.name,
      slug: model.slug,
      description: model.description,
      isActive: model.isActive,
      laundryServicesCount: model.laundryServicesCount,
      laundryServices: model.laundryServices
          ?.map((e) => LaundryService.fromModel(e))
          .toList(),
      createdAt: model.createdAt,
      updatedAt: model.updatedAt,
      deletedAt: model.deletedAt,
    );
  }

  @override
  List<Object?> get props => [
    id,
    outletId,
    name,
    slug,
    description,
    isActive,
    laundryServicesCount,
    laundryServices,
    createdAt,
    updatedAt,
    deletedAt,
  ];
}
